Makefile

```diff
@@ -940,12 +962,37 @@ endif
 	rm -f $(dir $@)/disk.raw
 	$(QUIET): $(dir $@)/$(notdir $*).img.tar.gz: Succeeded
 
+ifdef LIVE_UPDATE
+# Target depends on rootfs tarbar directly, which gives possibility to
+# detect when qcow2 should be updated with a tarball and when it should
+# be recreated from scratch.
+%.qcow2: %.raw $(ROOTFS_TAR) | $(DIST)
+#	Detect if the first %.raw ($<) prerequisite is in the "$?" list,
+#	which means qcow has to be fully recreated. If not - just update
+#	with the existing tar.
+	$(eval RECREATE := $(if $(filter $<,$?),1,0))
+#	Convert raw to qcow or update rootfs with all files from tar:
+#	guestfish one line magic.
+	$(QUIET)if [ "$(RECREATE)" = "1" ]; then \
```

Comment:
if you are just using RECREATE here, don't you think it's better to just integrate the checking to the if and remove the need of `eval`, like this `[  "$(if .....)" = "1" ]; then...`  ?

Reply:
I moved it to a separate flag not to overload the if branch and to have a separate description, what exactly the RECREATE flag means. I would keep it as is.